### PR TITLE
Summer 2020

### LINKED ISSUE #67: Update Dependencies for D3M
See https://gitlab.com/datadrivendiscovery/d3m/-/issues/445.

Do we have any issues with these?

### LINKED ISSUE #69: Submit Primitives For 2020 June Dry Run
- [x] Update core package version
- [x] Update dependencies to match [new D3M versions](https://gitlab.com/datadrivendiscovery/d3m/-/issues/445)
- [x] Update metalearn to match the new dependencies too
- [ ] Update copy of the datasets repo
- [ ] Bump the version of the primitives (if any code or dependency changes are made on them)
- [ ] Set D3M environment variables
- [ ] Run primitive pipelines on all seed datasets (MIN_METADATA)
- [ ] Submit primitives and all pipeline runs

### PR DESCRIPTION
**Pull request by epeters3**
Closes #69.
Closes #67.

### CODE REVIEW

**Comment by bjschoenfeld** [comment]
We need to do a release of metalearn that also has the updated dependencies.

**Comment by epeters3** [comment]
Thanks Brandon yes I saw that in the README.

**Comment by epeters3** [comment]
Oh nevermind, that's not in the README. Thank you for telling me that!

**Comment by epeters3** [comment]
@bjschoenfeld the pipeline runner is now working, although for the meantime its still leaving behind an un-gzipped version of each pipeline run that failed. Just letting you know in case you wanted to use the code to test the profiler primitive. Basically you'll need to add some basic metadata about your primitive to the `primitives_data` dictionary inside `submission.pipelines.generate_pipelines.main`. `main` will then generate a pipeline file for your primitive and try to run all the seed problems on it. Note that you'll need to specify a `"gen_method"` inside the `primitives_data` dictionary. The imputer and metafeatures primitive you may recall each have a method that can be called to generate a pipeline that uses that primitive. You'll need to define one for the profiler too.

**Comment by epeters3** [comment]
I am currently running the pipeline runner in this branch to run the pipelines for the imputer and metafeatures primitives on all seed problems. So far I have 3 successful pipeline run submissions for each primitive.

File: byudml/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-__version__ = '0.6.8'
+__version__ = '0.6.9'
 __metafeature_version__ = '0.4.5'
 __metafeature_path__ = 'd3m.primitives.metalearning.metafeature_extractor.BYU'
 __imputer_version__ = '0.2.1'
```
Comment (bjschoenfeld) [inline byudml/__init__.py:4]: Do we need to update the metafeature and imputer versions too? If there are any changes there, then we should.

Comment (epeters3) [reply byudml/__init__.py:4]: I have not needed to make any changes to either of the primitives. I did a bug fix to the imputer a couple weeks ago, but I already did a patch release on `byudml` for that, bumping the imputer version too (see 1ac5a7f216e0f064e9c65f9ea2bddf504c936e4d).

File: submission/pipelines/metafeature_extractor/baa68a80-3a7d-472d-8d4f-54918cc1bd8f.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
                 "version": "0.3.0",
                 "python_path": "d3m.primitives.data_transformation.dataset_to_dataframe.Common",
                 "name": "Extract a DataFrame from a Dataset",
-                "digest": "422744651afd5995d029a227a1dd7b1696038816b7eb9601f37d661757812aee"
+                "digest": "8457eeb19b44f5b26a5fb721a51087d618962a35f0aff17f4ad25d341a2170f2"
```
Comment (bjschoenfeld) [inline submission/pipelines/metafeature_extractor/baa68a80-3a7d-472d-8d4f-54918cc1bd8f.json:24]: Are you getting any digest warnings when you run things? We may need to remove these too.

Comment (epeters3) [reply submission/pipelines/metafeature_extractor/baa68a80-3a7d-472d-8d4f-54918cc1bd8f.json:24]: I did the first time I ran our tests a few days ago but I'm not anymore.

File: submission/pipelines/run_pipeline.py [hunk 3]
```diff
@@ -114,38 +117,45 @@ def run_pipeline_run(
...
         rerun_args += ['--scores', pipeline_rerun_scores_path]
-    cli.main(rerun_args)
+    try:
+        cli.main(rerun_args)
+    except Exception as e:
+        # Let's clean up after ourselves. We don't want to keep documents
+        # for pipeline runs that failed.
+        os.remove(pipeline_run_path)
+        os.remove(pipeline_rerun_path)
+        raise e
```
Comment (bjschoenfeld) [inline submission/pipelines/run_pipeline.py:137]: does this give us the real pipeline error or just some failure error?

File: byudml/__init__.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-__version__ = '0.6.8'
+__version__ = '0.6.9'
```
Comment (bjschoenfeld) [inline byudml/__init__.py:1]: I would wait to do this version bump until we are ready to merge devel into master

File: submission/pipelines/run_pipeline.py (hunk 3 above)
Comment (epeters3) [reply submission/pipelines/run_pipeline.py:137]: `raise e` reraises the original error, so we get the real error message. I'm just inserting a clean-up step in-between.

File: byudml/__init__.py (hunk 4 above)
Comment (epeters3) [reply byudml/__init__.py]: Done.

File: submission/pipelines/generate_pipelines.py [hunk 5]
```diff
@@ -598,21 +612,31 @@ def main():
...
             primitive_path = primitive.metadata.query()['python_path']
             submission_path = os.path.join(byu_dir, primitive_path, primitive_data['version'])
+            pipeline_run_name = f'{pipeline_json["id"]}_{problem.name}'
+            pipeline_run_path = os.path.join(submission_path, 'pipeline_runs', f"{pipeline_run_name}.yml.gz")
+            if os.path.isfile(pipeline_run_path):
+                print(
+                    f"pipeline {pipeline_json['id']} has already "
+                    f"been run on problem {problem.name}, skipping."
+                )
+                continue
```
Comment (bjschoenfeld) [inline submission/pipelines/generate_pipelines.py:637]: will this cause problems when we run things multiple times, say if it failed the first time?

File: byudml/__init__.py (hunk 1 above)
Comment (bjschoenfeld) [reply byudml/__init__.py]: The metafeature primitive has a changed dependency now on the metalearn package, so it should probably be bumped before we release.

**Pull request merged**